students/Chao/session04/trigram.py

```diff
@@ -1,5 +1,71 @@
 #!/usr/bin/env python
 
+import string
+import random
+
 # Create an empty dictionary
 trig = {}
+# An empty list for word pool
+wpool = []
+
+def build_pool(inf):
+    """ Build a pool of words from the input text file """
+    with open(inf, 'r') as lines:
+        # strip the line, make it all lower cases, remove punctuations, then make a new list
+        for line in lines:
+            wpool.extend(line.strip().lower().translate(str.maketrans('', '', string.punctuation)).split())
+
```

Comment:
Nice one liner. :) Note that because you are putting the entire file into your list, you will end up with stuff that isn't part of the actual story (headers, dates, intro stuff).